src/node_crypto.cc

```diff
@@ -745,6 +748,42 @@ void SecureContext::SetKey(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+#ifndef OPENSSL_NO_ENGINE
+void SecureContext::SetEngineKey(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  SecureContext* sc;
+  ASSIGN_OR_RETURN_UNWRAP(&sc, args.Holder());
+
+  unsigned int len = args.Length();
+  CHECK_EQ(len, 2);
+
+  char errmsg[1024];
+  const node::Utf8Value engine_id(env->isolate(), args[1]);
+  ENGINE* e = LoadEngineById(*engine_id, &errmsg);
+  if (e == nullptr) {
+    return env->ThrowError(errmsg);
+  }
+
+  if (!ENGINE_init(e)) {
```

Comment:
> To obtain a functional reference from an existing structural reference, call the ENGINE_init() function... otherwise it will return non-zero to indicate that the ENGINE is now operational and will have allocated a new functional reference to the ENGINE. All functional references are released by calling ENGINE_finish() (which removes the implicit structural reference as well).

Should there be a call to `ENGINE_finish()` somewhere?